RELEASES.md

```diff
@@ -278,6 +278,36 @@ and new fields on messages may be added if they are optional.
 to the API by having a diff that the CI can run. These files are not intended to be
 consumed or used by clients.
 
+As of containerd 2.0, the API version diverges from the main containerd version.
+While containerd 2.0 is a _major_ version jump for containerd, the API will remain
+on 1.x to remain backwards compatible with prior releases and existing clients.
+The 2.0 release adds the API to a separate Go module which can remain as the
+`github.com/containerd/containerd/api` Go package and imported separately from the
+rest of containerd.
+
+The API minor version will continue to be incremented for each major and minor
```

Comment:
I guess this depends on whether there were changes in the package / module? (currently it reads as it will always be incremented, which may not always be needed)

Reply:
(basically; versioning should depend on the changes to follow SemVer, which could mean that only a patch release is needed between containerd releases, and versions can be updated separate from containerd releases; order of releasing may still require the API to be tagged before doing a containerd release  of course)

Reply:
I'm sure there will always be changes, we can change the doc if a situation comes up later that doesn't make sense. 

It was an idea to remove, but I don't think that will be necessary. There are a few different ways we could handle the versioning, I think the cleanest will be just to not do patch releases for prior minor versions though. The logic is that the highest level API needs to be supported anyway and should not break any compatibility. The whole issue that made this change necessary is related to multiple Go packages in use and in that same scenario the API version will  go up. I feel like we have a few different ways we could approach the versioning though and this change gives us more options than we have without it.